src/lightning/pytorch/strategies/launchers/multiprocessing.py

```diff
@@ -97,6 +101,8 @@ def launch(self, function: Callable, *args: Any, trainer: Optional["pl.Trainer"]
         self._check_torchdistx_support()
         if self._start_method in ("fork", "forkserver"):
             _check_bad_cuda_fork()
+            if XPUAccelerator.is_available():
+                _check_bad_xpu_fork()
```

Comment:
would be better if check device first before calling corresponding fork api.

Reply:
Checks already present , additionally not needed.